ext/sockets/tests/gh16267.phpt

```diff
@@ -7,16 +7,16 @@ sockets
 --FILE--
 <?php
 try {
-	socket_strerror(PHP_INT_MIN);
+	socket_strerror(-2147483648);
```

Comment:
This has basically the same problem (32bit systems) as the bug we're trying to fix; the number `2147483648` overflows there.

Did you try:
````diff
 ext/sockets/sockets.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/sockets/sockets.c b/ext/sockets/sockets.c
index 08b8d8406d..5cc66f2d8f 100644
--- a/ext/sockets/sockets.c
+++ b/ext/sockets/sockets.c
@@ -370,7 +370,7 @@ char *sockets_strerror(int error) /* {{{ */
 
 #ifndef PHP_WIN32
 	if (error < -10000) {
-		error = -error - 10000;
+		error = (int)(unsigned int) error - 10000;
 
 #ifdef HAVE_HSTRERROR
 		buf = hstrerror(error);
````